assets/src/scripts/schema.js

```diff
@@ -72,7 +68,51 @@ const timeSeries = memoize(tsKey => new schema.Entity('timeSeries', {
 }));
 
 // timeSeriesCollection schema
-const queryInfo = memoize(tsKey => new schema.Entity('queryInfo', {}, {idAttribute: () => tsKey}));
+const queryInfo = memoize(tsKey => new schema.Entity('queryInfo', {}, {
+    idAttribute: () => tsKey,
+    processStrategy: value => {
+        const queryInfo = {
+            ...value,
+            notes: value.note.reduce((notes, note) => {
+                notes[note.title] = note.value;
+                return notes;
+            }, {})
+        };
+        delete queryInfo.note;
+
+        // Extract values out of filter:timeRange
+        // If this is a "period" query (eg, P7D)
+        if (queryInfo.notes['filter:timeRange'].indexOf('PERIOD') > -1) {
+            const regEx = /\[mode=(.+), period=P(.+)D, modifiedSince=(.+)\]/;
```

Comment:
It's too bad that this info is sent as a string containing an array rather than an array 😞